data/mods/gen1/scripts.ts

```diff
@@ -903,7 +903,8 @@ export const Scripts: ModdedBattleScriptsData = {
 			// Type effectiveness.
 			// In Gen 1, type effectiveness is applied against each of the target's types.
 			for (const targetType of target.types) {
-				const typeMod = this.battle.dex.getEffectiveness(type, targetType);
+				let typeMod = this.battle.dex.getEffectiveness(type, targetType);
+				typeMod = this.battle.runEvent('Effectiveness', this.battle, targetType, move, typeMod);
```

Comment:
Does `getEffectiveness` have a side effect? If not, you're just overwriting the value with `runEvent`'s return value.

Reply:
`getEffectiveness` outputs the (regular) type effectiveness; this is used as an input to `runEvent` which flips this value if Inverse Mod is active